src/main/java/com/aws/greengrass/mqttclient/spool/Spool.java

```diff
@@ -172,9 +178,12 @@ public void popOutMessagesWithQosZero() {
     private void removeMessagesWithQosZero(boolean needToCheckCurSpoolerSize) {
         Iterator<Long> messageIdIterator = queueOfMessageId.iterator();
         while (messageIdIterator.hasNext() && addJudgementWithCurrentSpoolerSize(needToCheckCurSpoolerSize)) {
-            long idToBeRemoved = messageIdIterator.next();
-            if (getMessageById(idToBeRemoved).getQos().getValue() == 0) {
-                removeMessageById(idToBeRemoved);
+            long id = messageIdIterator.next();
+            PublishRequest request = getMessageById(id);
+            if (request.getQos().getValue() == 0) {
+                removeMessageById(id);
+                logger.atDebug().kv("id", id).kv("topic", request.getTopic())
+                        .log("The message which Qos is 0 will be dropped when offline.");
```

Comment:
this message is a bit confusing because it isn't dropping the message in the future, it is dropping the message right now.

"The message has QoS 0 and the spooler is configured to drop QoS 0 when offline. Dropping message now."